lib/puppet_x/templates/init_ps.ps1.erb

```diff
@@ -476,7 +486,14 @@ function Invoke-PowerShellUserCode
       # is to return 1 so that we ensure Puppet reports this run as an error.
       $ec = 1
     }
-    $output = $_.Exception.Message | Out-String
+
+    if ($_.Exception.ErrorRecord.InvocationInfo -ne $null)
+    {
+      $output = $_.Exception.Message + "`n`r" + $_.Exception.ErrorRecord.InvocationInfo.PositionMessage
```

Comment:
And my mention of dislike is that they are multi-line, sometimes I find the output overly verbose when you are just trying to figure out what the error is. In this case it appears helpful though.